libs/vkd3d/command.c

```diff
@@ -2595,6 +2596,16 @@ static HRESULT STDMETHODCALLTYPE d3d12_command_list_Close(d3d12_command_list_ifa
     if (list->is_predicated)
         VK_CALL(vkCmdEndConditionalRenderingEXT(list->vk_command_buffer));
 
+    memory_barrier.sType = VK_STRUCTURE_TYPE_MEMORY_BARRIER;
+    memory_barrier.pNext = NULL;
+    memory_barrier.srcAccessMask = VK_ACCESS_MEMORY_WRITE_BIT;
```

Comment:
BOTTOM_OF_PIPE cannot do memory barriers. Need ALL_COMMANDS_BIT on both if we're going to emit memory barriers.